connectors/sources/notion.py

```diff
@@ -96,6 +96,48 @@ async def get_via_session(self, url):
             else:
                 raise
 
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=NotFound,
+    )
+    async def fetch_results(
+        self, function: Callable[..., Awaitable[Any]], next_cursor=None, **kwargs: Any
+    ):
+        try:
+            return await function(start_cursor=next_cursor, **kwargs)
+        except Exception as exception:
+            if exception.code == "rate_limited" and exception.status == 429:
+                retry_after = (
+                    exception.headers.get("retry-after") or DEFAULT_RETRY_SECONDS
+                )
+                self._logger.info(
```

Comment:
let's include in the logs what the request was. I assume that's captured in the `next_cursor` or the `kwargs`, but capturing the `function`'s name might be good too.